### DISCUSSION TITLE
Pin Changes when going for Pico W

### DISCUSSION THREAD

**Discussion by Myrmi123** [OpenGammaProject/Open-Gamma-Detector#58] [Q&A] [answered]
@NuclearPhoenixx : You mention several times that some internal pins change from the Pico to the Pico W board so the software needs to be modified. Which pins are this for example? I did a quick check and couldn't spot any difference.

**Comment by NuclearPhoenixx** [comment, marked as answer]
Hi there, there are a couple differences between internal pins.

Just had a quick google and copied text from the first result: [https://core-electronics.com.au/guides/raspberry-pi-pico-w-vs-pico-whats-the-difference/](https://core-electronics.com.au/guides/raspberry-pi-pico-w-vs-pico-whats-the-difference/)... Quote:

---

<html><body>
<div data-content-type="row" data-appearance="contained" data-element="main"><div data-enable-parallax="0" data-parallax-speed="0.5" data-background-images="{}" data-background-type="image" data-video-loop="true" data-video-play-only-visible="true" data-video-lazy-load="true" data-video-fallback-src="" data-element="inner" data-pb-style="LBKJDFB"><div data-content-type="text" data-appearance="default" data-element="main"><p id="DK6RJ91">Of
 course, this is the big one. The major upgrade in the Pico W is support
 for wireless connectivity (2.4GHz wireless LAN b/g/n). The wireless 
interface (Infineon CYW43439) is driven by pins internal to the Pico W 
(not broken out to the headers). These are:</p>
<ul style="list-style-type: disc;">
<li>GPIO29 (Input/Output) wireless SPI CLK/ADC mode (ADC3) measures VSYS/3</li>
<li>GPIO25 SPI CS (Output) when high also enables GPIO29 ADC pin to read VSYS</li>
<li>GPIO24 (Input/Output) wireless SPI data / IRQ</li>
<li>GPIO23 (Output) wireless power-on signal</li>
</ul>
<p>The wireless interface also brings a few new user-controllable pins to the mix:</p>
<ul style="list-style-type: disc;">
<li>WL_GPIO2 – (Input) VBUS sense. HIGH when VBUS is present</li>
<li>WL_GPIO1 – (Output) Controls onboard SMPS Power Save pin</li>
<li>WL_GPIO0 – (Output) Controls user LED</li>
</ul>
<p>This means that some internal pins on the Pico W are now repurposed 
for communication with the CYW43439, and have been replaced by GPIO 
present on the CY43439.</p>
<p>In summary, internal functions that have changed between models are:</p>

Function | Pico | Pico W
-- | -- | --
User LED | GPIO25 | WL_GPIO0
Power Supply PS | GPIO20 | WL_GPIO1
VBUS sense | GPIO24 | WL_GPIO2

</div></div></div><div data-content-type="row" data-appearance="contained" data-element="main"><div data-enable-parallax="0" data-parallax-speed="0.5" data-background-images="{}" data-background-type="image" data-video-loop="true" data-video-play-only-visible="true" data-video-lazy-load="true" data-video-fallback-src="" data-element="inner" data-pb-style="VCSY9DF"><h2 data-content-type="heading" data-appearance="default" data-element="main">On-board LED</h2><div data-content-type="text" data-appearance="default" data-element="main"><p>Looking
 closer at the Wireless section, we can see that the user-LED is now 
controlled by the wireless module on pin WL_GPIO0. On the original Pico 
this was GPIO25, which is now reserved for SPI communication with the 
wireless module. This difference will probably affect the most users, 
because the onboard LED is a convenient debugging tool / status 
indicator.</p></div></div></div>
</body>
</html>

**Comment by Myrmi123** [comment]
Ah, many thanks :)
Didn't see that. So VBUS cannot be measured anymore on the Pico W while using the wireless module. This shouldn't really matter as VBUS_MEAS is not necessary for operation according to your code. The LED can still be acessed through the wireless interface. I think I will order the Pico W and try to get it running.

**Reply by NuclearPhoenixx** [reply]
Yeah there aren't any important changes, so you should absolutely be able to use it with a Pico W by slightly altering the code. Glad I could help!
